### ISSUE TITLE
copyright agent: do not include characters that can not be part of a URL

### ISSUE THREAD

**Issue by yaobinshi** [fossology/fossology#317]
---

Author Name: **armijn hemel** 
Original Redmine Issue: 7245, http://www.fossology.org/issues/7245
Original Date: 2014/07/10

---

When processing the attached file from wxWidgets with the copyright agent in 2.5.0 I see characters that can never be part of a URL, such as a newline. If they were part of a URL they should have been escaped.

For example:

  [35788:35827:url] 'http://xml.libexpat.org/entity.ent'>\n"'
        [35864:35896:url] 'http://xml.libexpat.org/ns1'>\n"'
        [36492:36521:url] 'http://xml.libexpat.org/'>\n"'
        [37755:37784:url] 'http://xml.libexpat.org/'>\n"'
        [38033:38057:url] 'http://xml.libexpat.org/'
        [38421:38450:url] 'http://xml.libexpat.org/'>\n"'

What you can see is that there are single quotes, newlines, double quotes and right angle brackets that should not have been included.

**Comment by pguttmann** [comment]
Attachment - http://www.fossology.org/attachments/4389/runtests.c

**Comment by mcjaeger** [comment]
right, is still an issue (with 3.0.0). So there seem to be two solutions, a cheap one to match commonly known characters, such as " ' < > \ etc. or just take URL-safe characters and stop matching with the first non-URL-safe character.

**Comment by shaheemazmalmmd** [comment]
Tested with 4.0.0 issue still exist.

**Comment by SalmanDeveloperz** [comment]
I've successfully reproduced this issue on the latest master branch (commit: 7c0714f4c).

**Reproduction Steps:**
1. Created test file with:
```c
/* Test file */
const char *s =
  "http://xml.libexpat.org/entity.ent'>\n\"";
```
2. Uploaded test file through web UI
3. Ran copyright agent using "One-Shot Copyright/Email/URL Analysis"

**Current Behavior:**
The copyright agent includes invalid URL characters in the findings:
- Single quotes: `'`
- Newlines: `\n`
- Double quotes: `"`
- Angle brackets: `>`

**Expected Behavior:**
Should only capture: `http://xml.libexpat.org/entity.ent`

**Screenshot:**

<img width="1919" height="865" alt="Image" src="https://github.com/user-attachments/assets/d829b1db-da6f-4669-ba1c-e4abce45029d" />
